pkg/localmetrics/localmetrics.go

```diff
@@ -61,33 +63,47 @@ func UpdateAPIMetrics(APIkey string) {
 
 // UpdateMetricPagerDutyCreateFailure updates guage to 1 when creation fails
 func UpdateMetricPagerDutyCreateFailure(x int, cd string) {
-	MetricPagerDutyCreateFailure.With(prometheus.Labels{"name": "pagerduty-operator", "clusterdeployment_name": cd}).Set(float64(x))
+	MetricPagerDutyCreateFailure.With(prometheus.Labels{
+		"name": "pagerduty-operator", "clusterdeployment_name": cd}).Set(
+		float64(x))
 }
 
 // UpdateMetricPagerDutyDeleteFailure updates guage to 1 when deletion fails
 func UpdateMetricPagerDutyDeleteFailure(x int, cd string) {
-	MetricPagerDutyDeleteFailure.With(prometheus.Labels{"name": "pagerduty-operator", "clusterdeployment_name": cd}).Set(float64(x))
+	MetricPagerDutyDeleteFailure.With(prometheus.Labels{
+		"name": "pagerduty-operator", "clusterdeployment_name": cd}).Set(
+		float64(x))
 }
 
-// UpdateMetricPagerDutyHeartbeatGauge curls the PD API, updates the gauge to 1 when successful.
+// UpdateMetricPagerDutyHeartbeatGauge curls the PD API, updates the gauge to 1
+// when successful.
 func UpdateMetricPagerDutyHeartbeatGauge(APIkey string) {
 	metricLogger := log.WithValues("Namespace", "pagerduty-operator")
 	metricLogger.Info("Metrics for PD API")
 
 	_, err := http.NewRequest("GET", "api.pagerduty.com", nil)
 	if err != nil {
-		MetricPagerDutyHeartbeat.With(prometheus.Labels{"name": "pagerduty-operator"}).Set(float64(0))
+		MetricPagerDutyHeartbeat.With(prometheus.Labels{
+			"name": "pagerduty-operator"}).Set(float64(0))
 		metricLogger.Error(err, "Failed to get reach api")
 	}
 
 	// if there is an api key make an authenticated calld
 	if APIkey != "" {
-		client := pdApi.NewClient(APIkey)
-
-		if client != nil {
-			MetricPagerDutyHeartbeat.With(prometheus.Labels{"name": "pagerduty-operator"}).Set(float64(0))
-			metricLogger.Error(err, "Failed to get reach api when authenticated")
+		req, _ := http.NewRequest("GET", apiEndpoint, nil)
+		req.Header.Set("Accept", "application/vnd.pagerduty+json;version=2")
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Authorization", "Token token="+APIkey)
+
+		_, err := http.DefaultClient.Do(req)
+
+		if err != nil {
+			metricLogger.Error(err, "Failed to reach api when authenticated")
+			MetricPagerDutyHeartbeat.With(prometheus.Labels{
+				"name": "pagerduty-operator"}).Set(float64(0))
+			return
 		}
```

Comment:
You want to make sure you close your http requests after you use them otherwise they will not be closed and you'll introduce a resource leak with lots of open http connections that haven't been closed.

You will want to change the request to store the response `_, err` becomes `resp, err` and then add after the err check block `defer resp.Body.Close()`

Reply:
Done

Reply:
Done

Reply:
Done